base/shell.jl

```diff
@@ -81,7 +81,7 @@ function shell_parse(raw::AbstractString, interp::Bool)
                     if done(s,k)
                         error("unterminated double quote")
                     end
-                    if s[k] == '"' || s[k] == '$'
+                    if s[k] == '"' || s[k] == '$' || s[k] == '\\'
```

Comment:
This is a strict bug fix and unrelated to the rest of this change, so I've pulled it out and applied it here: https://github.com/JuliaLang/julia/commit/cd5922102189ffc329765dd60f038bf3109742b3.